service/id_validation.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/RedHatInsights/sources-api-go/config"
+	"github.com/labstack/echo/v4"
+)
+
+// ExtractValidateId attempts to extract the ID from the provided "paramName", and then performs a very simple "ID is
+// not empty" and "ID is greater than zero" checks.
+func ExtractValidateId(c echo.Context, idParamName string) (int64, error) {
+	idRaw := c.Param(idParamName)
+
+	if idRaw == "" {
+		return 0, errors.New("the provided ID cannot be empty or missing")
+	}
+
+	id, err := strconv.ParseInt(idRaw, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf(`could not parse the provided ID: %s`, err)
```

Comment:
can I ask what is difference between \` and " ?

Reply:
In this particular case it's true that it doesn't make much sense to use backticks, since I'm not wrapping the error message between quotes. This is mainly the reason I use backticks for, or when I want Golang to just not touch the string, and to keep the formatting I specify: https://go.dev/ref/spec#String_literals

I'm going to switch the `` ` `` for `"`.

Reply:
I see, thanks for explanation!